orchagent/routeorch.cpp

```diff
@@ -1012,13 +1012,16 @@ bool RouteOrch::addNextHopGroup(const NextHopGroupKey &nexthops)
 
     /* Assert each IP address exists in m_syncdNextHops table,
      * and add the corresponding next_hop_id to next_hop_ids. */
+    bool missingNexthop = false;
     for (auto it : next_hop_set)
     {
         if (!m_neighOrch->hasNextHop(it))
         {
-            SWSS_LOG_INFO("Failed to get next hop %s in %s",
+            SWSS_LOG_INFO("Failed to get next hop %s in %s, resolving neighbor",
                     it.to_string().c_str(), nexthops.to_string().c_str());
```

Comment:
This function suppose to create the NH group, should not trigger nbr resolution, please do the changes as follows in the function addRoute ()
```

/* Try to create a new next hop group */
            if (!addNextHopGroup(nextHops))
            {
                /* NextHopGroup is in "Ip1|alias1,Ip2|alias2,..." format*/
                std::vector<std::string> nhops = tokenize(nextHops.to_string(), ',');
                for(auto it = nhops.begin(); it != nhops.end(); ++it)
                {
                    NextHopKey nextHop;
                    if (overlay_nh) {
                        nextHop = NextHopKey(*it, overlay_nh);
                    } else {
                        nextHop = NextHopKey(*it);
                    }

                    if(!m_neighOrch->hasNextHop(nextHop))
                    {
/* Add the nbr resolution here */
....
....
}
```

Reply:
Thanks for this suggestion. Fixed as prescribed.

Reply:
ok